projects/canopy/src/lib/forms/sort-code/sort-code.directive.ts

```diff
@@ -0,0 +1,40 @@
+import { Directive, HostBinding, HostListener, OnInit } from '@angular/core';
+import { NgControl, Validators } from '@angular/forms';
+
+@Directive({
+  selector: '[lgSortCode]',
+})
+export class LgSortCodeDirective implements OnInit {
+  @HostBinding('placeholder') placeholder = '00-00-00';
+  @HostBinding('required') required = true;
+  @HostBinding('attr.inputmode') inputmode = 'numeric';
+  // 8 because we allow for the two dashes
+  @HostBinding('attr.maxlength') maxlength = '8';
+  @HostBinding('attr.size') size = '11';
+  @HostListener('focusout', ['$event.target.value']) onBlur(value) {
+    if (this.ngControl.valid) {
+      this.ngControl.valueAccessor.writeValue(this.format(value));
+    }
+  }
+
+  constructor(private ngControl: NgControl) {}
+
+  ngOnInit(): void {
+    const validators = [
+      Validators.required,
+      Validators.pattern(/^(?:\d{6}|\d\d-\d\d-\d\d)$/),
+    ];
+
+    if (this.ngControl.control.validator) {
+      validators.push(this.ngControl.control.validator);
+    }
+    this.ngControl.control.setValidators(validators);
+  }
+
+  private format(value: string): string {
+    return value
+      .replace(/\-/g, '')
+      .match(/.{1,2}/g)
+      .join('-');
```

Comment:
I might be reading this wrong, but it is first replacing all hyphens with a space, then added them back in again? Do we need the first bit?

I'm also wondering if we should allow users to enter spaces, as well as hyphens? I think that's also a common way of typing a sort code.

Again sorry if I missed something, regex isn't my strong suit!

Reply:
I've changed the code to allow single spaces now 👍 

Yes, that's right. We remove them and then we add them back. This is for when a user enters directly `00-00-00` and the formatter would transform it to `00--00--00`.

Reply:
Ah yes of course, makes sense now. Thanks for adding the spaces.